haskell-src/Concordium/Types/Queries.hs

```diff
@@ -0,0 +1,278 @@
+      -- |Details of the finalization record in the block (if any)
+      bsFinalizationData :: Maybe FinalizationSummary,
+      -- |Details of the update queues and chain parameters as of the block
+      bsUpdates :: Updates
+    }
+
+$(deriveJSON defaultOptions{fieldLabelModifier = firstLower . dropWhile isLower} ''BlockSummary)
+
+data RewardStatus = RewardStatus
+    { -- |The total GTU in existence
+      rsTotalAmount :: Amount,
+      -- |The total GTU in encrypted balances
+      rsTotalEncryptedAmount :: Amount,
+      -- |The amount in the baking reward account
+      rsBakingRewardAccount :: Amount,
+      -- |The amount in the finalization reward account
+      rsFinalizationRewardAccount :: Amount,
+      -- |The amount in the GAS account
+      rsGasAccount :: Amount
+    }
+
+$(deriveJSON defaultOptions{fieldLabelModifier = firstLower . dropWhile isLower} ''RewardStatus)
+
+-- |Summary of a baker.
+data BakerSummary = BakerSummary
+    { -- |Baker ID
+      bsBakerId :: BakerId,
+      -- |(Approximate) lottery power
+      bsBakerLotteryPower :: Double,
+      -- |Baker account (should never be @Nothing@)
+      bsBakerAccount :: Maybe AccountAddress
+    }
+
+$(deriveJSON defaultOptions{fieldLabelModifier = firstLower . dropWhile isLower} ''BakerSummary)
+
+-- |Summary of the birk parameters applicable to a particular block.
+data BlockBirkParameters = BlockBirkParameters
+    { -- |Baking lottery election difficulty
+      bbpElectionDifficulty :: ElectionDifficulty,
+      -- |Current leadership election nonce for the lottery
+      bbpElectionNonce :: LeadershipElectionNonce,
+      -- |List of the currently eligible bakers
+      bbpBakers :: Vec.Vector BakerSummary
+    }
+
+$(deriveJSON defaultOptions{fieldLabelModifier = firstLower . dropWhile isLower} ''BlockBirkParameters)
+
+-- |The status of a transaction that is present in the transaction table.
+data TransactionStatus
+    = -- |Transaction was received but is not in any blocks
+      Received
+    | -- |Transaction was received an is present in some (non-finalized) block(s)
+      Committed (Map.Map BlockHash (Maybe TransactionSummary))
+    | -- |Transaction has been finalized in a block
+      Finalized BlockHash (Maybe TransactionSummary)
+
+instance ToJSON TransactionStatus where
+    toJSON Received = object ["status" .= String "received"]
+    toJSON (Committed m) =
+        object
+            [ "status" .= String "finalized",
+              "outcomes" .= m
+            ]
+    toJSON (Finalized bh outcome) =
+        object
+            [ "status" .= String "finalized",
+              "outcomes" .= Map.singleton bh outcome
+            ]
+
+-- |The status of a transaction with respect to a specified block
+data BlockTransactionStatus
+    = -- |Either the transaction is not in that block, or that block is not live
+      BTSNotInBlock
+    | -- |The transaction was received but not known to be in that block
```

Comment:
The documentation is unclear which of the two statuses above would make sense if a transaction is received, but not in a given block.

Reply:
Both are possible. If the transaction is received but not in any block, then the status will be `BTSReceived`. If the transaction is received but in a different block, then the status will be `BTSNotInBlock`. Note that at the RPC level, `BTSNotInBlock` is indistinguishable from the case where the transaction is completely absent.

Reply:
I'm aware of the possibilities, my complaint was that the documentation of the options is ambiguous and should be clarified.

Reply:
I don't think I understand what the intended semantics of `getTransactionStatusInBlock` is, then. Perhaps you could document this? Do we rely somewhere on the distinction between these cases?